arwen/host/execution.go

```diff
@@ -253,13 +255,15 @@ func (host *vmHost) finishExecuteOnDestContext(gasUsed uint64, executeErr error)
 	return vmOutput
 }
 
-func (host *vmHost) ExecuteOnSameContext(input *vmcommon.ContractCallInput) (asyncInfo *arwen.AsyncContextInfo, err error) {
+func (host *vmHost) ExecuteOnSameContext(input *vmcommon.ContractCallInput) (*arwen.AsyncContext, error) {
 	log.Trace("ExecuteOnSameContext", "function", input.Function)
 
 	if host.IsBuiltinFunctionName(input.Function) {
 		return nil, arwen.ErrBuiltinCallOnSameContextDisallowed
 	}
 
+	var err error
```

Comment:
no need for this var. at line 285 you can have err :=

Reply:
Declaration is required here due to its use in `defer`.